fiat-roles/src/main/java/com/netflix/spinnaker/fiat/config/ResourceProvidersHealthIndicator.java

```diff
@@ -34,27 +36,51 @@
 
   @Autowired @Setter List<HealthTrackable> providers;
 
+  @Autowired @Setter Registry registry;
+
   private AtomicBoolean previousHealthCheckIsUp = new AtomicBoolean(false);
 
+  private AtomicBoolean upOnce = new AtomicBoolean(false);
+
+  private Id id(String name, HealthTrackable provider) {
+    return registry
+        .createId("fiat.resourceProvider." + name)
+        .withTag("provider", provider.getClass().getSimpleName());
+  }
+
   @Override
   protected void doHealthCheck(Health.Builder builder) throws Exception {
     boolean isDown = false;
     for (HealthTrackable provider : providers) {
       builder.withDetail(
           provider.getClass().getSimpleName(), provider.getHealthTracker().getHealthView());
       isDown = isDown || !provider.getHealthTracker().isProviderHealthy();
+      registry
+          .gauge(id("healthy", provider))
```

Comment:
doesn't this create a new gauge with the same id on every `doHealthcheck` call? I think this is a legitimate use case for `PolledMeter` instead

Reply:
you can safely do `registry.gauge("foo).set(value)` repeatedly - it doesn't allocate a new object each time, just tracks the one measurement per id